### PR TITLE
move to rebar3 and update healthcheck logic

### PR DESCRIPTION
**Pull request by id**

### CODE REVIEW

File: src/brucke_routes.erl [hunk 1]
```diff
@@ -38,9 +39,11 @@
...
 -spec init([raw_route()]) -> ok | no_return().
 init(Routes) when is_list(Routes) ->
-  ets:info(?ETS) =/= ?undef andalso exit({?ETS, already_created}),
-  ?ETS = ets:new(?ETS, [named_table, protected, set,
-                        {keypos, #route.upstream}]),
+  ets:info(?T_ROUTES) =/= ?undef andalso exit({?T_ROUTES, already_created}),
+  ets:info(?T_DISCARDED_ROUTES) =/= ?undef andalso exit({?T_DISCARDED_ROUTES, already_created}),
+  ets:new(?T_ROUTES, [named_table, protected, set,
+                      {keypos, #route.upstream}]),
+  ets:new(?T_DISCARDED_ROUTES, [named_table, protected, bag]),
```
Comment (zmstone) [inline src/brucke_routes.erl:45]: duplicate_bag ?

Comment (id) [reply src/brucke_routes.erl:45]: should we? It won't help much to see 2 identical routes were discarded

Comment (zmstone) [reply src/brucke_routes.erl:45]: right

File: .travis.yml [hunk 2]
```diff
@@ -20,15 +20,12 @@ otp_release:
...
-  - make app
-  - make xref
-  - make edoc
-  - make test-env
-  - make t
-  - make distclean
-  - echo "verifying rebar3 compilation"
   - git clone https://github.com/erlang/rebar3.git; cd rebar3; ./bootstrap; cd ..
-  - ./rebar3/rebar3 compile
+  - rebar=./rebar3/rebar3 make vsn-check
```
Comment (zmstone) [inline .travis.yml]: export rebar=... ?

File: rebar.config [hunk 3]
```diff
@@ -5,4 +5,49 @@
...
+           , warn_obsolete_guard
+           , debug_info
+           ]}.
+
+{ct_use_short_names, true}.
+{cover_enabled, true}.
+{eunit_opts, [verbose]}.
+
+{relx, [ {release, {brucke, "version is taken from VSN file"},
+          [sasl, brucke, brod, supervisor3, kafka_protocol]}
```
Comment (zmstone) [inline rebar.config]: no need to list brod, supervisor3 and kafka_protocol

File: src/brucke_http.erl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                              , {onresponse, fun error_hook/4}]),
+  ok.
+
+error_hook(Code, _Headers, _Body, Req) ->
+  Method = cowboy_req:method(Req),
+  Version = cowboy_req:version(Req),
+  Path = cowboy_req:path(Req),
+  {Ip, _Port} = cowboy_req:peer(Req),
+  Level = log_level(Code),
+  lager:log(Level, self(), "~s ~s ~b ~p ~s", [Method, Path, Code, Version, inet:ntoa(Ip)]),
```
Comment (zmstone) [inline src/brucke_http.erl]: FORMAT ERROR: "~s ~s ~b ~p ~s" [{<<"GET">>,{http_req,#Port<0.1020>,ranch_tcp,keepalive...

File: src/brucke_http.erl [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                                     , {"/healthcheck", brucke_http_healthcheck_handler, []}
+                                     ]}]),
+  lager:info("Starting http listener on port ~p", [Port]),
+  {ok, _} = cowboy:start_http(http, 8, [{port, Port}],
+                              [ {env, [{dispatch, Dispatch}]}
+                              , {onresponse, fun error_hook/4}]),
+  ok.
+
+error_hook(Code, _Headers, _Body, Req) ->
+  Method = cowboy_req:method(Req),
```
Comment (zmstone) [inline src/brucke_http.erl]: {Method, NewReq} = ... ?

File: src/brucke_http.erl (hunk 4 above)
Comment (id) [reply src/brucke_http.erl]: oops

**Review by zmstone — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

**Pull request merged**